src/execution_plan/ops/op_node_by_label_scan.c

```diff
@@ -77,23 +130,30 @@ static Record NodeByLabelScanConsumeFromChild(OpBase *opBase) {
 		if(op->child_record == NULL) return NULL;
 		// One-time construction of the iterator.
 		// (Don't do so before now because the label might have been built in a child op.)
-		if(op->iter == NULL) _ConstructIterator(op);
-		else GxB_MatrixTupleIter_reset(op->iter);
+		if(op->iter == NULL) {
+			if(!_ConstructIterator(op)) {
+				// No need to iterate - consume child.
+				while(OpBase_Consume(op->op.children[0])) {}
+				return NULL;
+			}
+		} else {
+			_ResetIterator(op);
+		}
 	}
 
 	GrB_Index nodeId;
 	bool depleted = false;
 	GxB_MatrixTupleIter_next(op->iter, NULL, &nodeId, &depleted);
-	if(depleted) {
+	if(depleted || _outOfBounds(op, nodeId)) {
 		Record_Free(op->child_record); // Free old record.
 		// Pull a new record from child.
 		op->child_record = OpBase_Consume(op->op.children[0]);
 		if(op->child_record == NULL) return NULL; // Child depleted.
 
 		// Reset iterator and evaluate again.
-		GxB_MatrixTupleIter_reset(op->iter);
+		_ResetIterator(op);
 		GxB_MatrixTupleIter_next(op->iter, NULL, &nodeId, &depleted);
-		if(depleted) return NULL; // Empty iterator; return immediately.
+		if(depleted || _outOfBounds(op, nodeId)) return NULL; // Empty iterator; return immediately.
```

Comment:
I diss agree I believe we should continue pulling, although the iterator did not managed to produce data, we still need to continue pulling data, this will allow all of the previous operation to run.

Reply:
fixed